include/deal.II/fe/fe_p1nc.h

```diff
@@ -0,0 +1,316 @@
+ *
+ * Unlike any continuous conforming finite element belongs to H^1_0,
+ * the P1 nonconforming element does not enforce the continuity across edges.
+ * But it requires the continuity just in integral sense:
+ * any function in the space should have the same integral values on two sides of the common edge shared by two adjacent elements.
+
+ * Thus each function in the nonconforming element space can be discontinuous, not included in H^1_0, as functions in Discontinuous Galerkin (DG) finite element spaces.
+ * Although any function in DG element space also has nonconformity, it is completely discontinuous across edges without any relation.
+ * It is a reason why usual weak formulations for DG schemes contain additional penalty terms for jump across edges to control discontinuity.
+ * However nonconforming elements usually do not need additional terms in their weak formulations due to the continuity in integral on edges.
+
+ * <h3>DOFs and Dice Rule</h3>
+ * Since any function in the P1 nonconforming space is piecewise linear on each element,
+ * the function value at the mipoint of each edge is same to the mean value on the edge.
+ * Thus the continuity of the integral value across each edge is equivalent to
+ * the continuity of the midpoint value of each edge in this case.
+ *
+ * The degrees of freedom (DOFs) on a quadrilateral are defined by midpoint values on edges.
+ * But these four DOFs are not independent in fact.
+ * The simple observation reads that any linear function on a quadrilateral
+ * satisfies 'dice rule': the sum of two function values at two midpoints of the edge pair on opposite
+ * position is equal to the sum of those of the another edge pair.
+ *
+ * \phi(m_0) + \phi(m_1) = \phi(m_2) + \phi(m_3).
+ *
+ * Conversely if 4 values at midpoints satisfying the dice rule are just given,
+ * then there always exists the unique linear function which coincides with 4 midpoints values.
+ *
+ * Due to the dice rule, three values at any three midpoints can determine the last value at the last midpoint.
+ * It means that the genuine number of (independent) DOFs on a quad is 3,
+ * and it is the same number to the dimension of the linear polynomial space in 2D.
+
+
+ * <h3>Shape functions</h3>
+
+ *  @verbatim
+ *  2---------|---------3
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  -                   -
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  |                   |
+ *  0---------|---------1
+ *  @endverbatim
+
+ * For each vertex v_j of given quad, there are two edges of which v_j is one of end points.
+ * Consider a linear function such that 0.5 value at two midpoints of such edges,
+ * and 0.0 at two midpoints of other edges.
+ * Note that the set of these values satisfies the dice rule which is described above.
+ * We denote such a function assoicated with vertex v_j by \phi_j.
```

Comment:
assoicated -> associated
Don't forget to use $ when you have a mathematical expression